### PR TITLE
Move code around: Refactor and rename filters and metadata interfaces

### PR DESCRIPTION
**Pull request by dougbu**
- part of #178 and #179
- related to #181 because Trello receiver needs secret key confirmation but no challenge
- `WebHookGetResponseFilter` -> `WebHookGetRequestFilter`
  - unconditionally confirm secret key exists
  - stop using `WebHookResultUtilities` in this class
- `WebHookPingResponseFilter` -> `WebHookPingRequestFilter`
- `WebHookVerifyBodyContentFilter` -> `WebHookVerifySignatureFilter` and use it less
  - stop using `EncodingUtilities` or `WebHookResultUtilities` in this class
- `StripeTestEventResponseFilter` -> `StripeTestEventRequestFilter`
- `IWebHookRequestMetadata` -> `IWebHookBodyTypeMetadata`
- `IWebHookRequestMetadataService` -> `IWebHookBodyTypeMetadataSerivce`
- split `IWebHookEventMetadata` into that plus `IWebHookPingRequestMetadata`
- split `IWebHookSecurityMetadata` into `IWebHookGetRequestMetadata` and `IWebHookVerifyCodeMetadata`
  - remove `WebHookGetRequest` class

Also
- remove `EncodingUtilities`
- refactor `WebHookMetadataProvider` while updating metadata validations
- improve comments, logging, and error messages in changed files

### CODE REVIEW

**Comment by dougbu** [comment]
@rynowak I recommend looking at the metadata interfaces before skimming through the receiver-specific bits. Lots of pattern matching to be had here.

**Comment by dougbu** [comment]
@rynowak have you had 👀 on this :one: ?

File: src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookMetadataProvider.cs [hunk 1]
```diff
@@ -129,7 +106,8 @@ private void Apply(ActionModel action)
             var receiverName = attribute.ReceiverName;
             if (receiverName != null)
             {
-                var bindingMMetadata = _bindingMetadata.FirstOrDefault(metadata => metadata.IsApplicable(receiverName));
+                var bindingMMetadata = _bindingMetadata.FirstOrDefault(
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookMetadataProvider.cs]: typo bindingMMetadata

**Comment by dougbu** [comment]
:up::date: mostly for AppVeyor run&hellip;

**Comment by dougbu** [comment]
93ea86c77f

**Pull request closed without merging**